### ISSUE TITLE
Missing superimport module for compare_results.ipynb

### ISSUE THREAD

**Issue by Neoanarika** [probml/pyprobml#662]
I encounter the following error when running compare_results.ipynb in the vae folder.

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-2-d04a2cc14755> in <module>()
      5 import matplotlib.pyplot as plt
      6 import torchvision.transforms as transforms
----> 7 from download_celeba import celeba_dataloader
      8 from assembler import get_config, assembler
      9 from utils.plot import plot_reconstruction, plot_samples

/content/download_celeba.py in <module>()
----> 1 import superimport
      2 
      3 from absl import app
      4 from absl import flags
      5 import torchvision.transforms as transforms
```

I think this can be resolved by adding one line in the notebook to download superimport into the colab.

**Comment by mjsML** [comment]
@Neoanarika which notebook exactly are you talking about?

**Comment by Neoanarika** [comment]
I was referring to the colab link in the readme of the VAE subfolder but the file itself is in the probml-notebooks repository. https://colab.research.google.com/github/probml/probml-notebooks/blob/main/notebooks/vae_compare_results.ipynb. I guessed I should have opened this issue in the notebook repository.

**Comment by gerdm** [comment]
@Neoanarika, I think `pip install superimport` should fix the problem. Every script in pyrprobml now depends on `superimport`

**Comment by Neoanarika** [comment]
Yup, that is what I eventually did. Thanks, @gerdm. I will make a pull request to fix this issue.

**Comment by Neoanarika** [comment]
Ok, I noticed something, although  ```pip install superimport``` worked for gan_compare_results.ipynb file. It didn't work for vae_compare_results.ipynb. Instead, I get this error when trying to run download_celeba.py

```
**raceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/superimport/superimport.py", line 228, in <module>
    import_module(file_name,package, True)
  File "/usr/local/lib/python3.7/dist-packages/superimport/superimport.py", line 129, in import_module
    _dir,package=get_dir(file_name)
  File "/usr/local/lib/python3.7/dist-packages/superimport/superimport.py", line 124, in get_dir
    package=file_path.split("/")[-2]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "download_celeba.py", line 1, in <module>
    import superimport
  File "/usr/local/lib/python3.7/dist-packages/superimport/__init__.py", line 1, in <module>
    from superimport import superimport
  File "/usr/local/lib/python3.7/dist-packages/superimport/superimport.py", line 244, in <module>
    install_if_missing({package}, True)
  File "/usr/local/lib/python3.7/dist-packages/superimport/superimport.py", line 65, in install_if_missing
    subprocess.check_call([python3, "-m", "pip", "install", *missing])
  File "/usr/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'pip', 'install', 'absl']' returned non-zero exit status 1.**
```

Any idea what is going on @gerdm ?

**Comment by gerdm** [comment]
@Neoanarika ,
I'm pretty sure it's because super import is trying to install the `absl` library, when it should be `absl-py`

```
(test) ❯ pip install absl                                                                                                                                                    ~
ERROR: Could not find a version that satisfies the requirement absl (from versions: none)
ERROR: No matching distribution found for absl

(test) 1 ❯ pip install absl-py                                                                                                                                               ~
Collecting absl-py
  Using cached absl_py-1.0.0-py3-none-any.whl (126 kB)
Collecting six
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: six, absl-py
Successfully installed absl-py-1.0.0 six-1.16.0
```

Maybe `pip install absl-py` at the top of the notebook should help? What do you think, @mjsML?

**Comment by murphyk** [comment]
I moved Ming's 2 GAN notebooks to pyprobml/gan from pyprobml-notebooks.
I added the `pip install superimport` to gan_compare_results.

 https://github.com/probml/pyprobml/blob/master/gan/gan_compare_results.ipynb 
https://github.com/probml/pyprobml/blob/master/gan/gan_tricks.ipynb.

**Comment by murphyk** [comment]
I (partially) fixed the VAE demo. I put the new code at https://github.com/probml/pyprobml/blob/master/vae/vae_compare_results.ipynb, the version in probml-notebooks is deprecated. The problem had nothing to do with absl (which is pre-installed). The problem was that the `import superimport` statement that was added to https://github.com/probml/pyprobml/blob/master/scripts/download_celeba.py broke the functionality, since that script assumes there is a local file called [data.py](https://github.com/probml/pyprobml/blob/master/vae/data.py) on the path, so that `from data import  CelebADataModule` works correctly. If you have superimport, it tries to download a package called `data` from pypi, which is the wrong module.

The notebook now (mostly) runs. However there is an error in the sampling cell, which calls this line:
 ```
plot_samples(vaes, num=num_imgs, num_of_images_per_row=num_of_images_per_row, figdir=figdir)
```
This causes the error `mat1 and mat2 shapes cannot be multiplied (6x256 and 64x150)`.
@Neoanarika Please can you debug this. (I assume it's a batch size issue where we asked for 6 samples but somewhere the code has hard-coded the assumption that batch size=64.)

**Comment by murphyk** [comment]
It turns out that Ming's original https://github.com/probml/probml-notebooks/blob/main/notebooks/gan_compare_results.ipynb works as is, without needing any changes. This is because it does not call `download_celeba.py`. (This is because only the VAE code shows reconstructions of original images, the GAN code just samples.) I have removed the version of the notebooks that I added to probml/gan, and left the original  ones in their probml-notebooks location, since that is what the book (and README) expects. Sorry for the confusion.
